File: exercise/day02/src/FizzBuzz.ts
import { OutOfRangeException } from './OutOfRangeException';

export class FizzBuzz {
  private constructor() {}

  public static convert(input: number): string {
    if (input > 0) {
      if (input <= 100) {
        if (input % 3 === 0 && input % 5 === 0) {
          return 'FizzBuzz';
        }
        if (input % 3 === 0) {
          return 'Fizz';
        }
        if (input % 5 === 0) {
          return 'Buzz';
        }
        return input.toString();
      } else {
        throw new OutOfRangeException();
      }
    } else {
      throw new OutOfRangeException();
    }
  }
}

File: solution/kotlin/day06/src/main/kotlin/games/FizzBuzz.kt
package games

private const val MIN = 0
private const val MAX = 100

object FizzBuzz {
    fun convert(input: Int): String = when {
        isOutOfRange(input) -> throw OutOfRangeException()
        `is`(15, input) -> "FizzBuzz"
        `is`(3, input) -> "Fizz"
        `is`(5, input) -> "Buzz"
        else -> input.toString()
    }

    private fun `is`(divisor: Int, input: Int): Boolean {
        return input % divisor == 0
    }

    private fun isOutOfRange(input: Int) = input <= MIN || input > MAX
}
